src/state/slices/opportunitiesSlice/opportunitiesSlice.ts

```diff
@@ -0,0 +1,117 @@
+import { createSlice } from '@reduxjs/toolkit'
+import { createApi } from '@reduxjs/toolkit/query/react'
+import type { AccountId, AssetId } from '@shapeshiftoss/caip'
+import type { DefiType } from 'features/defi/contexts/DefiManagerProvider/DefiCommon'
+import { BASE_RTK_CREATE_API_CONFIG } from 'state/apis/const'
+
+export const initialState: OpportunitiesState = {
+  lp: {
+    byAccountId: {},
+    byId: {},
+    ids: [],
+  },
+  farming: {
+    byAccountId: {},
+    byId: {},
+    ids: [],
+  },
+  userLp: {
+    byId: {},
+    ids: [],
+  },
+  userFarming: {
+    byId: {},
+    ids: [],
+  },
+}
+
+// The contract info, also referred to as "metadata" - doesn't contain any user-specific data
+type Opportunity = {
+  // The LP token AssetId
+  assetId: AssetId
+  provider: string
+  tvl: string
+  apy: string
+  type: DefiType
+}
+
+type LpOpportunity = Opportunity & {
+  // A tuple of the underlying assets this LP represents
+  underlyingAssetIds: [AssetId, AssetId]
+}
+
+type FarmingOpportunity = Opportunity
+
+// User-specific values for this opportunity
+type UserLpOpportunity = LpOpportunity & {
+  // The amount of held LP tokens
+  cryptoAmount: string
+  underlyingFoxCryptoBaseUnitAmount: string
+  underlyingEthCryptoBaseUnitAmount: string
+}
+
+// User-specific values for this opportunity
+type UserFarmingOpportunity = FarmingOpportunity & {
+  // The amount of farmed LP tokens
+  cryptoAmount: string
+}
+
+// The AccountId of the farming contract in the form of chainId:accountAddress
+type FarmingId = AccountId
+// The AccountId of the LP contract in the form of chainId:accountAddress
+type LpId = AccountId
+// The unique identifier of an lp opportunity in the form of UserAccountId:FarmingId
```

Comment:
i know we haven't implemented these ids yet - but we may want to consider using `*` instead of `:` as a delimiter like we do for our tx ids, then we can easily split into two ids

https://github.com/shapeshift/web/blob/develop/src/state/slices/txHistorySlice/utils.ts#L48

Reply:
write these serializer/deserializer functions, and some dummy selector to check the ergonomics of this works for us and solves what we're trying to solve